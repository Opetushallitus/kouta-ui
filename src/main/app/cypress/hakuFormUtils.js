import merge from 'lodash/merge';

import organisaatio from './data/organisaatio';
import {
  stubKoodistoRoute,
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
} from './utils';

export const stubHakuFormRoutes = ({ cy, organisaatioOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  stubKoodistoRoute({ koodisto: 'haunkohdejoukko', cy });
  stubKoodistoRoute({ koodisto: 'hakutapa', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });
  stubKoodistoRoute({ koodisto: 'opetuspaikkakk', cy });
  stubKoodistoRoute({ koodisto: 'kausi', cy });

  cy.route({
    method: 'GET',
    url: '**/haku/list**',
    response: [],
  });

  stubHakemuspalveluLomakkeetRoute({ cy });
  stubOppijanumerorekisteriHenkiloRoute({ cy });
};
