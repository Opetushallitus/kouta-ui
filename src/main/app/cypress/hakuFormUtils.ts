import createKoodisto from '#/cypress/data/koodisto';

import organisaatio from '#/cypress/data/organisaatio';
import createKoodi from '#/cypress/data/koodi';

import {
  stubKoodistoRoute,
  stubHakemuspalveluLomakkeetRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubKoodiRoute,
  stubCommonRoutes,
} from './utils';

export const stubHakuFormRoutes = ({ organisaatioOid }) => {
  stubCommonRoutes();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: organisaatio({
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      organisaatio({
        oid: organisaatioOid,
      }),
    ],
  });

  stubKoodistoRoute({ koodisto: 'haunkohdejoukontarkenne' });
  stubKoodistoRoute({ koodisto: 'kausi' });
  stubKoodistoRoute({ koodisto: 'opetuspaikkakk' });
  stubKoodistoRoute({ koodisto: 'kausi' });
  stubKoodistoRoute({ koodisto: 'posti' });
  stubKoodiRoute(createKoodi({ koodisto: 'posti', versio: 2 }));

  cy.route({
    method: 'GET',
    url: `**/koodisto-service/rest/json/haunkohdejoukko/koodi**`,
    response: createKoodisto({ koodisto: 'haunkohdejoukko' }).map((v, index) =>
      index === 0
        ? {
            ...v,
            koodiUri: 'haunkohdejoukko_12',
          }
        : v
    ),
  });

  cy.route({
    method: 'GET',
    url: `**/koodisto-service/rest/json/hakutapa/koodi**`,
    response: createKoodisto({ koodisto: 'hakutapa' }).map((v, index) =>
      index === 0
        ? {
            ...v,
            koodiUri: 'hakutapa_01',
          }
        : v
    ),
  });

  cy.route({
    method: 'GET',
    url: '**/haku/list**',
    response: [],
  });

  cy.route({
    method: 'GET',
    url: `**/toteutus/list**`,
    response: [],
  });

  stubHakemuspalveluLomakkeetRoute();
  stubOppijanumerorekisteriHenkiloRoute();
};
