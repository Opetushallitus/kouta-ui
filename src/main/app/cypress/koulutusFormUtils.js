import merge from 'lodash/merge';

import { stubKoodistoRoute } from './utils';
import organisaatioHierarkia from './data/organisaatioHierarkia';
import organisaatio from './data/organisaatio';
import koodisto from './data/koodisto';

export const stubKoulutusFormRoutes = ({ cy, organisaatioOid }) => {
  cy.server();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae?oid=${organisaatioOid}**`,
    response: organisaatioHierarkia({ rootOid: organisaatioOid }),
  });

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-ylakoodit/koulutustyyppi_*',
    response: koodisto({ koodisto: 'koulutus' }),
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/list?organisaatioOid=${organisaatioOid}`,
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_0',
    response: [
      {
        koodiUri: 'koulutus_0',
        versio: 1,
        metadata: [{ kieli: 'fi', nimi: 'Nimi' }],
      },
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/eperusteet-service/api/perusteet?tuleva=true&siirtyma=false&voimassaolo=true&poistunut=false&kieli=fi&koulutuskoodi=koulutus_0',
    response: {
      data: [
        {
          kuvaus: { fi: 'koulutus_0 kuvaus' },
          osaamisalat: [],
          tutkintonimikeKoodiUri: 'nimike_1#1',
          id: '1',
        },
      ],
    },
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_0**',
    response: [
      {
        koodiUri: 'okmohjauksenala_1',
        metadata: [{ kieli: 'fi', nimi: 'Ala' }],
      },
      {
        koodiUri: 'opintojenlaajuus_1',
        metadata: [{ kieli: 'fi', nimi: '180' }],
      },
      {
        koodiUri: 'opintojenlaajuusyksikko_1',
        metadata: [{ kieli: 'fi', nimi: 'op' }],
      },
    ],
  });

  stubKoodistoRoute({ koodisto: 'tutkintonimikekk', cy });
  stubKoodistoRoute({ koodisto: 'opintojenlaajuus', cy });
  stubKoodistoRoute({ koodisto: 'koulutuksenjarjestamisenlisaosiot', cy });
};
