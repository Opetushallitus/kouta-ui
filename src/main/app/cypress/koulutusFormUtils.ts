import { merge } from 'lodash';

import {
  stubKoodistoRoute,
  stubOppijanumerorekisteriHenkiloRoute,
  stubEPerusteetByKoulutuskoodiRoute,
  stubCommonRoutes,
} from '#/cypress/utils';
import organisaatioHierarkia from '#/cypress/data/organisaatioHierarkia';
import organisaatio from '#/cypress/data/organisaatio';
import koodisto from '#/cypress/data/koodisto';
import koodistoKoulutus351107 from './data/koodistoKoulutus351107';
import koodistoKoulutukset from './data/koodistoKoulutukset';
import koodistoKoulutus351107Alakoodit from './data/koodistoKoulutus351107Alakoodit';
import koodistoKoulutus309902 from './data/koodistoKoulutus309902';
import koodistoKoulutus309902Alakoodit from './data/koodistoKoulutus309902Alakoodit';
import koodistoKoulutus671112 from './data/koodistoKoulutus671112';
import koodistoKoulutus671112Alakoodit from './data/koodistoKoulutus671112Alakoodit';
import ePerustePerusteenosat6777674 from './data/ePerustePerusteenosat6777674';
import ePerustePerusteenosat6778360 from './data/ePerustePerusteenosat6778360';
import ePeruste6777660TutkinnonOsat from './data/ePeruste6777660TutkinnonOsat';
import ePeruste6777660Rakenne from './data/ePeruste6777660Rakenne';
import ePeruste6777660 from './data/ePeruste6777660';

export const stubKoulutusFormRoutes = ({ organisaatioOid }) => {
  stubCommonRoutes();

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/hierarkia/hae**`,
    response: organisaatioHierarkia({ rootOid: organisaatioOid }),
  });

  cy.route({
    method: 'GET',
    url: `**/organisaatio-service/rest/organisaatio/v4/${organisaatioOid}**`,
    response: merge(organisaatio(), {
      oid: organisaatioOid,
    }),
  });

  cy.route({
    method: 'POST',
    url: '**/organisaatio-service/rest/organisaatio/v4/findbyoids',
    response: [
      merge(organisaatio(), {
        oid: organisaatioOid,
      }),
    ],
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-ylakoodit/koulutustyyppi_*',
    response: koodisto({ koodisto: 'koulutus' }),
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/list?organisaatioOid=${organisaatioOid}`,
    response: [],
  });

  cy.route({
    method: 'GET',
    url: `**/koulutus/*/toteutukset/list**`,
    response: [],
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_351107/**',
    response: koodistoKoulutus351107,
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_351107**',
    response: koodistoKoulutus351107Alakoodit,
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_309902/**',
    response: koodistoKoulutus309902,
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_309902**',
    response: koodistoKoulutus309902Alakoodit,
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/codeelement/koulutus_671112/**',
    response: koodistoKoulutus671112,
  });

  cy.route({
    method: 'GET',
    url:
      '**/koodisto-service/rest/json/relaatio/sisaltyy-alakoodit/koulutus_671112**',
    response: koodistoKoulutus671112Alakoodit,
  });

  cy.route({
    method: 'GET',
    url: '**/koodisto-service/rest/json/koulutus/koodi?onlyValidKoodis=true**',
    response: koodistoKoulutukset,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteenosat/6777674',
    response: ePerustePerusteenosat6777674,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteenosat/6778360',
    response: ePerustePerusteenosat6778360,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteenosat/6778360',
    response: ePerustePerusteenosat6778360,
  });

  cy.route({
    method: 'GET',
    url: '**/tutkinnonosat',
    response: ePeruste6777660TutkinnonOsat,
  });

  cy.route({
    method: 'GET',
    url:
      '**/eperusteet-service/api/perusteet/6777660/suoritustavat/reformi/rakenne',
    response: ePeruste6777660Rakenne,
  });

  cy.route({
    method: 'GET',
    url: '**/eperusteet-service/api/perusteet/6777660',
    response: ePeruste6777660,
  });

  stubEPerusteetByKoulutuskoodiRoute();

  stubKoodistoRoute({ koodisto: 'tutkintonimikekk' });
  stubKoodistoRoute({ koodisto: 'opintojenlaajuus' });
  stubKoodistoRoute({ koodisto: 'koulutuksenlisatiedot' });

  stubKoodistoRoute({
    koodisto: 'kansallinenkoulutusluokitus2016koulutusalataso2',
  });

  stubOppijanumerorekisteriHenkiloRoute();
};
