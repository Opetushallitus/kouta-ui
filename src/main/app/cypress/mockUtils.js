const path = require('path');

const MOCK_PATH = 'cypress/mocks';

export const addMockFileRoutes = fileName => {
  cy.readFile(path.join(MOCK_PATH, fileName)).then(data => {
    data.forEach(({ url, method = 'GET', response }) => {
      // TODO: Mock only on matching response body and use cy.intercept()
      cy.route({
        method,
        url,
        status: response.status,
        response: response.body,
      });
    });
  });
};
