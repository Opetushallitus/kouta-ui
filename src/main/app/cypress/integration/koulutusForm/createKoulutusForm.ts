import { playMocks } from 'kto-ui-common/cypress/mockUtils';

import { stubKoulutusFormRoutes } from '#/cypress/koulutusFormUtils';
import koulutusMocks from '#/cypress/mocks/koulutus.mocks.json';
import {
  getSelectOption,
  getCheckbox,
  selectOption,
  fillTreeSelect,
  fillAsyncSelect,
  getByTestId,
  paste,
  fillKieliversiotSection,
  fillPohjaSection,
  fillTilaSection,
  tallenna,
  fillKoulutustyyppiSection,
  typeToEditor,
  wrapMutationTest,
  withinSection,
  getSelectByLabel,
  pFillSelect,
  getInputByLabel,
  pFillAsyncSelect,
} from '#/cypress/utils';
import { ENTITY } from '#/src/constants';

const fillLisatiedotSection = () => {
  withinSection('lisatiedot', () => {
    getByTestId('osiotSelect').click();

    getByTestId('osiotSelect').within(() => {
      getSelectOption('Opintojen rakenne').click({
        force: true,
      });
    });

    getByTestId('osioKuvaus.koulutuksenlisatiedot_01#1').within(() => {
      typeToEditor('koulutuksenlisatiedot kuvaus');
    });
  });
};

const fillSoraKuvausSection = () => {
  withinSection('soraKuvaus', () => {
    selectOption('Sora-kuvaus 1');
  });
};

const fillJarjestajaSection = () => {
  withinSection('tarjoajat', () => {
    getByTestId('jarjestajatSelection').within(() => {
      fillTreeSelect(['1.2.1.1.1.1']);
    });
  });
};

const fillCommon = ({ koulutustyyppiPath }) => {
  fillKoulutustyyppiSection(koulutustyyppiPath);
  fillPohjaSection();
  fillKieliversiotSection({ jatka: true });
};

const fillNakyvyysSection = () => {
  withinSection('julkinen', () => {
    getCheckbox(null).click({ force: true });
  });
};

export const createKoulutusForm = () => {
  const organisaatioOid = '1.1.1.1.1.1';
  const koulutusOid = '1.2.3.4.5';
  beforeEach(() => {
    stubKoulutusFormRoutes({ organisaatioOid });
    playMocks(koulutusMocks);

    cy.visit(`/organisaatio/${organisaatioOid}/koulutus`);
  });

  const mutationTest = wrapMutationTest({
    oid: koulutusOid,
    entity: ENTITY.KOULUTUS,
  });

  it(
    'should be able to create ammatillinen koulutus',
    mutationTest(() => {
      fillCommon({ koulutustyyppiPath: ['amm'] });

      withinSection('information', () => {
        getByTestId('koulutusSelect').click();

        getByTestId('koulutusSelect').within(() => {
          fillAsyncSelect('Kaivosalan perustutkinto');
        });

        getByTestId('ePerusteSelect').within(() => {
          fillAsyncSelect('Kaivosalan perustutkinto');
        });
      });

      fillLisatiedotSection();

      fillSoraKuvausSection();

      fillJarjestajaSection();

      fillNakyvyysSection();

      fillTilaSection();

      tallenna();
    })
  );

  it(
    'should be able to create ammatillinen osaamisala koulutus',
    mutationTest(() => {
      fillCommon({ koulutustyyppiPath: ['ammatillinen', 'amm-osaamisala'] });

      withinSection('osaamisala', () => {
        getByTestId('koulutusSelect').click();

        getByTestId('koulutusSelect').within(() => {
          fillAsyncSelect('Kaivosalan perustutkinto');
        });

        getByTestId('ePerusteSelect').within(() => {
          fillAsyncSelect('Kaivosalan perustutkinto');
        });

        getByTestId('osaamisalaSelect').within(() => {
          fillAsyncSelect('Kaivostyön osaamisala');
        });

        cy.findByRole('link', { name: '1800' }).should($link => {
          const url = new URL($link.attr('href'));
          expect(url.pathname).to.equal('/');
          expect(url.hash).to.equal(
            '#/fi/esitys/6777660/reformi/sisalto/6858226'
          );
        });
      });

      fillLisatiedotSection();

      fillSoraKuvausSection();

      fillJarjestajaSection();

      fillNakyvyysSection();

      fillTilaSection();

      tallenna();
    })
  );

  it(
    'should be able to create ammatillinen tutkinnon osa koulutus',
    mutationTest(() => {
      fillCommon({ koulutustyyppiPath: ['ammatillinen', 'amm-tutkinnon-osa'] });

      withinSection('tutkinnonosat', () => {
        getByTestId('lisaaKoulutusButton').click();

        getSelectByLabel('yleiset.valitseKoulutus').pipe(
          pFillAsyncSelect('Kaivosalan perustutkinto')
        );

        getByTestId('ePerusteSelect').within(() => {
          fillAsyncSelect('Kaivosalan perustutkinto');
        });

        getByTestId('tutkinnonOsatSelect').within(() => {
          fillAsyncSelect('Louhintaporaus');
        });

        cy.findByRole('link', { name: '106436' }).should($link => {
          const url = new URL($link.attr('href'));
          expect(url.pathname).to.equal('/');
          expect(url.hash).to.equal(
            '#/fi/esitys/6777660/reformi/tutkinnonosat/6778201'
          );
        });
      });

      withinSection('nimi', () => {
        getInputByLabel('koulutuslomake.lisaaKoulutuksenNimi').should(
          'have.value',
          'Louhintaporaus'
        );
      });

      withinSection('tutkinnonosat', () => {
        getByTestId('tutkinnonOsatSelect').within(() => {
          fillAsyncSelect('Kaivosmittaus');
        });

        cy.findByRole('link', { name: '106442' }).should($link => {
          const url = new URL($link.attr('href'));
          expect(url.pathname).to.equal('/');
          expect(url.hash).to.equal(
            '#/fi/esitys/6777660/reformi/tutkinnonosat/6778207'
          );
        });
      });

      fillLisatiedotSection();

      fillSoraKuvausSection();

      fillJarjestajaSection();

      fillNakyvyysSection();

      fillTilaSection();

      tallenna();
    })
  );

  it(
    'should be able to create AMK-koulutus',
    mutationTest(() => {
      fillCommon({ koulutustyyppiPath: ['korkeakoulutus', 'amk'] });

      withinSection('information', () => {
        getSelectByLabel('yleiset.valitseKoulutukset').pipe(
          pFillAsyncSelect('Fysioterapeutti (AMK)')
        );

        getSelectByLabel('koulutuslomake.valitseOpintojenLaajuus').pipe(
          pFillSelect('300')
        );

        getSelectByLabel('koulutuslomake.valitseTutkintonimike').pipe(
          pFillAsyncSelect('Fysioterapeutti (AMK)')
        );

        getSelectByLabel('koulutuslomake.valitseKoulutusalat').pipe(
          pFillAsyncSelect('Terveys')
        );

        getInputByLabel('koulutuslomake.muokkaaKoulutuksenNimea').pipe(
          paste('Tiedot nimi')
        );
      });

      withinSection('description', () => {
        getInputByLabel('yleiset.kuvauksenNimi').pipe(paste('Kuvauksen nimi'));

        getInputByLabel('yleiset.kuvaus').pipe(paste('Kuvaus'));
      });

      fillLisatiedotSection();

      fillSoraKuvausSection();

      fillJarjestajaSection();

      fillTilaSection();

      tallenna();
    })
  );

  it(
    'should be able to create korkeakoulutus opintojakso koulutus',
    mutationTest(() => {
      fillCommon({
        koulutustyyppiPath: ['korkeakoulutus', 'kk-opintojakso'],
      });

      withinSection('information', () => {
        getByTestId('laajuusnumero').pipe(paste('30'));

        getSelectByLabel('yleiset.laajuusyksikko').pipe(
          pFillSelect('opintopistettä')
        );

        getSelectByLabel('koulutuslomake.valitseKoulutusalat').pipe(
          pFillAsyncSelect('Terveys')
        );

        getInputByLabel('koulutuslomake.koulutuksenNimi').pipe(
          paste('Opintojakso nimi')
        );
      });

      withinSection('description', () => {
        getInputByLabel('yleiset.kuvauksenNimi').pipe(paste('Kuvauksen nimi'));

        getInputByLabel('yleiset.kuvaus').pipe(paste('Kuvaus'));
      });

      fillLisatiedotSection();

      fillJarjestajaSection();

      fillTilaSection();

      tallenna();
    })
  );

  it(
    'should be able to create Ammatillinen opettaja- erityisopettaja ja opokoulutus',
    mutationTest(() => {
      fillCommon({
        koulutustyyppiPath: ['korkeakoulutus', 'amm-ope-erityisope-ja-opo'],
      });

      withinSection('information', () => {
        getSelectByLabel('yleiset.valitseKoulutus').pipe(
          pFillAsyncSelect('Ammatillinen opettajankoulutus')
        );

        getInputByLabel('koulutuslomake.valitseOpintojenLaajuus')
          .should('be.disabled')
          .should('have.value', '60 opintopistettä');

        getInputByLabel('koulutuslomake.valitseTutkintonimike')
          .should('be.disabled')
          .should('have.value', 'koulutuslomake.eiTutkintonimiketta');

        getSelectByLabel('koulutuslomake.valitseKoulutusalat').should(
          'have.text',
          'kansallinenkoulutusluokitus2016koulutusalataso1_01#1'
        );

        getInputByLabel('koulutuslomake.muokkaaKoulutuksenNimea').should(
          'have.value',
          'Ammatillinen opettajankoulutus'
        );
      });

      withinSection('description', () => {
        getInputByLabel('yleiset.kuvauksenNimi').pipe(paste('Kuvauksen nimi'));

        getInputByLabel('yleiset.kuvaus').pipe(paste('Kuvaus'));
      });

      fillLisatiedotSection();

      fillSoraKuvausSection();

      fillJarjestajaSection();

      fillTilaSection();

      tallenna();
    })
  );

  it(
    'should be able to create lukiokoulutus',
    mutationTest(() => {
      fillCommon({ koulutustyyppiPath: ['lk'] });

      withinSection('information', () => {
        getSelectByLabel('yleiset.valitseKoulutus').pipe(
          pFillAsyncSelect('Ylioppilastutkinto')
        );
      });

      withinSection('description', () => {
        getInputByLabel('yleiset.kuvaus').pipe(paste('Kuvaus'));
      });

      fillLisatiedotSection();

      fillSoraKuvausSection();

      fillJarjestajaSection();

      fillTilaSection();

      tallenna();
    })
  );

  it(
    'should be able to create TUVA-koulutus',
    mutationTest(() => {
      fillCommon({ koulutustyyppiPath: ['tuva'] });

      withinSection('information', () => {
        getSelectByLabel('koulutuslomake.valitseOpintojenLaajuus').pipe(
          pFillSelect('38 viikkoa')
        );

        getInputByLabel('koulutuslomake.koulutuksenNimi').should(
          'have.value',
          'koulutustyypit.tuva'
        );
      });

      withinSection('description', () => {
        getInputByLabel('koulutuslomake.kuvauksenNimi').should('not.exist');

        getInputByLabel('yleiset.kuvaus').pipe(paste('Kuvaus'));

        getInputByLabel('koulutuslomake.linkkiEPerusteisiin').pipe(
          paste('http://linkki.fi')
        );
      });

      fillJarjestajaSection();

      fillTilaSection();

      getByTestId('soraKuvausSection').should('not.exist');
      getByTestId('lisatiedotSection').should('not.exist');

      tallenna();
    })
  );

  it(
    'should be able to create "Vapaa Sivistystyö - Opistovuosi"-koulutus',
    mutationTest(() => {
      fillCommon({
        koulutustyyppiPath: [
          'vapaa-sivistystyo',
          'vapaa-sivistystyo-opistovuosi',
        ],
      });

      withinSection('information', () => {
        getSelectByLabel('koulutuslomake.valitseOpintojenLaajuus').pipe(
          pFillSelect('Vähintään 53 op')
        );

        getInputByLabel('koulutuslomake.koulutuksenNimi').pipe(
          paste('vapaa sivistystyö nimi')
        );
      });

      withinSection('description', () => {
        getInputByLabel('koulutuslomake.kuvauksenNimi').should('not.exist');

        getInputByLabel('yleiset.kuvaus').pipe(paste('Kuvaus'));

        getInputByLabel('koulutuslomake.linkkiEPerusteisiin').pipe(
          paste('http://linkki.fi')
        );
      });

      fillJarjestajaSection();

      fillTilaSection();

      getByTestId('soraKuvausSection').should('not.exist');
      getByTestId('lisatiedotSection').should('not.exist');

      tallenna();
    })
  );

  it(
    'should be able to create "Vapaa Sivistystyö - Muu"-koulutus',
    mutationTest(() => {
      fillCommon({
        koulutustyyppiPath: ['vapaa-sivistystyo', 'vapaa-sivistystyo-muu'],
      });

      withinSection('information', () => {
        getSelectByLabel('koulutuslomake.valitseOpintojenLaajuus').pipe(
          pFillSelect('Vähintään 53 op')
        );

        getInputByLabel('koulutuslomake.koulutuksenNimi').pipe(
          paste('vapaa sivistystyö nimi')
        );
      });

      withinSection('description', () => {
        getInputByLabel('koulutuslomake.kuvauksenNimi').should('not.exist');

        getInputByLabel('yleiset.kuvaus').pipe(paste('Kuvaus'));

        getInputByLabel('koulutuslomake.linkkiEPerusteisiin').pipe(
          paste('http://linkki.fi')
        );
      });

      fillJarjestajaSection();

      fillTilaSection();

      getByTestId('soraKuvausSection').should('not.exist');
      getByTestId('lisatiedotSection').should('not.exist');

      tallenna();
    })
  );

  it(
    'should be able to create TELMA-koulutus',
    mutationTest(() => {
      fillCommon({ koulutustyyppiPath: ['ammatillinen', 'telma'] });

      withinSection('information', () => {
        getSelectByLabel('koulutuslomake.valitseOpintojenLaajuus').pipe(
          pFillSelect('60')
        );

        getInputByLabel('koulutuslomake.koulutuksenNimi').should(
          'have.value',
          'koulutustyypit.telma'
        );
      });

      withinSection('description', () => {
        getInputByLabel('koulutuslomake.kuvauksenNimi').should('not.exist');

        getInputByLabel('yleiset.kuvaus').pipe(paste('Kuvaus'));

        getInputByLabel('koulutuslomake.linkkiEPerusteisiin').pipe(
          paste('http://linkki.fi')
        );
      });

      fillJarjestajaSection();

      fillNakyvyysSection();

      fillTilaSection();

      getByTestId('soraKuvausSection').should('not.exist');
      getByTestId('lisatiedotSection').should('not.exist');

      tallenna();
    })
  );

  it(
    'should be able to create muu ammatillinen koulutus',
    mutationTest(() => {
      fillCommon({ koulutustyyppiPath: ['ammatillinen', 'amm-muu'] });

      withinSection('information', () => {
        getSelectByLabel('yleiset.laajuusyksikko').pipe(
          pFillSelect('osaamispistettä')
        );

        getByTestId('laajuusnumero').pipe(paste('12'));

        getInputByLabel('koulutuslomake.koulutuksenNimi').pipe(
          paste('muu ammatillinen nimi')
        );
      });

      withinSection('description', () => {
        getInputByLabel('koulutuslomake.kuvauksenNimi').should('not.exist');

        getInputByLabel('yleiset.kuvaus').pipe(paste('Kuvaus'));
      });

      fillJarjestajaSection();

      fillTilaSection();

      getByTestId('linkkiEPerusteisiinInput').should('not.exist');
      getByTestId('soraKuvausSection').should('not.exist');
      getByTestId('lisatiedotSection').should('not.exist');

      tallenna();
    })
  );

  it(
    'should be able to create "Aikuisten perusopetus" -koulutus',
    mutationTest(() => {
      fillCommon({ koulutustyyppiPath: ['aikuisten-perusopetus'] });

      withinSection('information', () => {
        getSelectByLabel('yleiset.laajuusyksikko').pipe(
          pFillSelect('opintopistettä')
        );

        getByTestId('laajuusnumero').pipe(paste('13'));

        getInputByLabel('koulutuslomake.koulutuksenNimi').should(
          'have.value',
          'koulutustyypit.aikuistenPerusopetus'
        );
      });

      withinSection('description', () => {
        getInputByLabel('koulutuslomake.kuvauksenNimi').should('not.exist');

        getInputByLabel('yleiset.kuvaus').pipe(paste('Kuvaus'));

        getInputByLabel('koulutuslomake.linkkiEPerusteisiin').pipe(
          paste('http://linkki.fi')
        );
      });

      fillJarjestajaSection();

      fillNakyvyysSection();

      fillTilaSection();

      getByTestId('soraKuvausSection').should('not.exist');
      getByTestId('lisatiedotSection').should('not.exist');

      tallenna();
    })
  );

  it(
    'should be able to create EB-koulutus',
    mutationTest(() => {
      fillCommon({ koulutustyyppiPath: ['lk'] });

      withinSection('information', () => {
        getSelectByLabel('yleiset.valitseKoulutus').pipe(
          pFillAsyncSelect('EB-tutkinto (European Baccalaureate)')
        );

        getSelectByLabel('koulutuslomake.valitseOpintojenLaajuus').pipe(
          pFillSelect('60')
        );

        getSelectByLabel('koulutuslomake.valitseKoulutusalat').should(
          'have.text',
          'Yleissivistävä koulutus'
        );

        getInputByLabel('koulutuslomake.muokkaaKoulutuksenNimea').should(
          'have.value',
          'EB-tutkinto (European Baccalaureate)'
        );
      });

      withinSection('description', () => {
        getInputByLabel('yleiset.kuvaus').pipe(paste('Kuvaus'));
      });

      fillLisatiedotSection();

      fillSoraKuvausSection();

      fillJarjestajaSection();

      fillTilaSection();

      tallenna();
    })
  );
};
