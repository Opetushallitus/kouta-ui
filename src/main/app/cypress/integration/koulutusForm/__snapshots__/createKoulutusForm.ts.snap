exports[`createKoulutusForm > should be able to create ammatillinen koulutus #0`] =
{
  "ePerusteId": 1,
  "johtaaTutkintoon": true,
  "julkinen": true,
  "kielivalinta": [
    "fi"
  ],
  "koulutusKoodiUri": "koulutus_0#1",
  "koulutustyyppi": "amm",
  "metadata": {
    "koulutusalaKoodiUrit": [],
    "kuvauksenNimi": {},
    "kuvaus": {},
    "lisatiedot": [
      {
        "otsikkoKoodiUri": "koulutuksenlisatiedot_0#1",
        "teksti": {
          "fi": "koulutuksenlisatiedot_0 kuvaus"
        }
      }
    ],
    "opintojenLaajuusKoodiUri": null,
    "tutkintonimikeKoodiUrit": [],
    "tyyppi": "amm"
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "koulutus_0"
  },
  "organisaatioOid": "1.1.1.1.1.1",
  "tarjoajat": [
    "1.2.1.1.1.1"
  ],
  "teemakuva": null,
  "tila": "julkaistu"
};

exports[`createKoulutusForm > should be able to create korkeakoulu koulutus #0`] =
{
  "johtaaTutkintoon": true,
  "julkinen": true,
  "kielivalinta": [
    "fi"
  ],
  "koulutusKoodiUri": "koulutus_0#1",
  "koulutustyyppi": "yo",
  "metadata": {
    "koulutusalaKoodiUrit": [
      "kansallinenkoulutusluokitus2016koulutusalataso2_0#1"
    ],
    "kuvauksenNimi": {
      "fi": "Kuvauksen nimi"
    },
    "kuvaus": {
      "fi": "Kuvaus"
    },
    "lisatiedot": [
      {
        "otsikkoKoodiUri": "koulutuksenlisatiedot_0#1",
        "teksti": {
          "fi": "koulutuksenlisatiedot_0 kuvaus"
        }
      }
    ],
    "opintojenLaajuusKoodiUri": "opintojenlaajuus_0#1",
    "tutkintonimikeKoodiUrit": [
      "tutkintonimikekk_0#1"
    ],
    "tyyppi": "yo"
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "Tiedot nimi"
  },
  "organisaatioOid": "1.1.1.1.1.1",
  "tarjoajat": [
    "1.2.1.1.1.1"
  ],
  "teemakuva": null,
  "tila": "julkaistu"
};

exports[`createKoulutusForm > should be able to create lukiokoulutus #0`] =
{
  "johtaaTutkintoon": true,
  "julkinen": true,
  "kielivalinta": [
    "fi"
  ],
  "koulutusKoodiUri": "koulutus_0#1",
  "koulutustyyppi": "lk",
  "metadata": {
    "koulutusalaKoodiUrit": [],
    "kuvauksenNimi": {},
    "kuvaus": {},
    "lisatiedot": [
      {
        "otsikkoKoodiUri": "koulutuksenlisatiedot_0#1",
        "teksti": {
          "fi": "koulutuksenlisatiedot_0 kuvaus"
        }
      }
    ],
    "opintojenLaajuusKoodiUri": null,
    "tutkintonimikeKoodiUrit": [],
    "tyyppi": "lk"
  },
  "muokkaaja": "1.2.246.562.24.62301161440",
  "nimi": {
    "fi": "koulutus_0"
  },
  "organisaatioOid": "1.1.1.1.1.1",
  "tarjoajat": [
    "1.2.1.1.1.1"
  ],
  "teemakuva": null,
  "tila": "julkaistu"
};
